### PR TITLE
feat: add data layer to DifferentialDriveMap

### PR DESCRIPTION
**Pull request by bot190**
Adds data layer for the DifferentialDriveMap that exposes motor parameters, set point, and gyro angle.

### CODE REVIEW

**Comment by msoucy** [comment]
Should revise after #175

**Comment by bot190** [comment]
Thanks for updating this. We should merge this instead of letting it sit for another year.

**Comment by msoucy** [comment]
You don't want to throw it a birthday party? 🤣

File: core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java [hunk 1]
```diff
@@ -55,14 +52,15 @@ public void reset() {
...
+        this.getData().right.setpoint = 0.0;
     }
 
     @Override
     public void periodic() {
         super.periodic();
-        this.odometry.update(this.getRotation(), this.map.leftEncoder().getDistance(),
-                this.map.rightEncoder().getDistance());
+        this.odometry.update(this.getRotation(), this.getData().left.distanceInches,
+                this.getData().left.distanceInches);
```
Comment (bot190) [inline core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java:63]:
```suggestion
                this.getData().right.distanceInches);
```

**Review by msoucy — APPROVED** [review]

**Pull request merged**